federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -54,6 +58,13 @@ import (
 	"k8s.io/client-go/discovery"
 )
 
+const (
+	apiserverWaitTimeout   = 30 * 2 * time.Second
+	apiserverRetryInterval = 1 * time.Second
+
+	federationOnlyNamespace = "federation"
```

Comment:
This needs to be configurable, and should be an explicitly reserved name.  As it stands right now, if the user has a namespace called federation in any of their underlying clusters, these will be automatically deleted (because what you're saying here is that you only want it in the federation, not in any of the underlying clusters).